parley/src/layout/data.rs

```diff
@@ -133,30 +143,62 @@ pub struct LineRunData {
     pub text_range: Range<usize>,
     /// Range of clusters.
     pub cluster_range: Range<usize>,
-    /// Advance for the run.
-    pub advance: f32,
 }
 
-impl LineRunData {
+impl LineItemData {
+    pub fn is_text_run(&self) -> bool {
+        self.kind == LayoutItemKind::TextRun
+    }
+
+    pub fn is_inline_box(&self) -> bool {
+        self.kind == LayoutItemKind::InlineBox
+    }
+
     pub fn compute_line_height<B: Brush>(&self, layout: &LayoutData<B>) -> f32 {
-        let mut line_height = 0f32;
-        let glyph_start = layout.runs[self.run_index].glyph_start;
-        for cluster in &layout.clusters[self.cluster_range.clone()] {
-            if cluster.glyph_len != 0xFF && cluster.has_divergent_styles() {
-                let start = glyph_start + cluster.glyph_offset as usize;
-                let end = start + cluster.glyph_len as usize;
-                for glyph in &layout.glyphs[start..end] {
-                    line_height = line_height.max(layout.styles[glyph.style_index()].line_height);
+        match self.kind {
+            LayoutItemKind::TextRun => {
+                let mut line_height = 0f32;
+                let run = &layout.runs[self.index];
+                let glyph_start = run.glyph_start;
+                for cluster in &layout.clusters[run.cluster_range.clone()] {
+                    if cluster.glyph_len != 0xFF && cluster.has_divergent_styles() {
+                        let start = glyph_start + cluster.glyph_offset as usize;
+                        let end = start + cluster.glyph_len as usize;
+                        for glyph in &layout.glyphs[start..end] {
+                            line_height =
+                                line_height.max(layout.styles[glyph.style_index()].line_height);
+                        }
+                    } else {
+                        line_height = line_height
+                            .max(layout.styles[cluster.style_index as usize].line_height);
+                    }
                 }
-            } else {
-                line_height =
-                    line_height.max(layout.styles[cluster.style_index as usize].line_height);
+                line_height
+            }
+            LayoutItemKind::InlineBox => {
+                // TODO: account for vertical alignment (e.g. baseline alignment)
+                layout.inline_boxes[self.index].height
```

Comment:
So we don't implement flowing around the inline boxes; that makes sense - it seems like it would be a nightmare to do so.

But we should probably make sure that's documented somewhere